### PR TITLE
WIP: Support parse the header line

### PR DESCRIPTION
**Pull request by zztttt**
## What type of PR is this：
- [ ] bug
- [·] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：
CSV load parser Support parse the header line #4026

## Problem Summary(Required) ：
Provided the custom parameter support about -H "skip_headline=true" for the HTTP request of stream load. I modified the thrift data structure between fe and be, and thus indicate whether the CSV headline can be ignored.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 4 / 9 (44.44%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/task/StreamLoadTask.java|3|5|60.00%|
|:large_blue_circle:|com/starrocks/planner/StreamLoadPlanner.java|1|1|100.00%|

File: be/src/exec/vectorized/file_scan_node.h [hunk 1]
```diff
@@ -59,10 +59,10 @@ class FileScanNode final : public ScanNode {
     }
 
     // Create scanners to do scan job
-    Status _start_scanners();
+    Status _start_scanners(bool skip_headline);
```
Comment (meegoo) [inline be/src/exec/vectorized/file_scan_node.h:62]: should move skip_headline to TBrokerScanRangeParams, you can refer to the implementation of multi_column_separator

File: be/src/exec/vectorized/file_scan_node.cpp [hunk 2]
```diff
@@ -285,14 +289,22 @@ void FileScanNode::_scanner_worker(int start_idx, int length) {
...
+            for (TBrokerRangeDesc range_desc : scan_range.ranges) {
                 // file_size is optional, and is not set in stream load and routine load,
                 // so we should check file size is set firstly.
                 if (range_desc.__isset.file_size && range_desc.file_size == 0) {
                     continue;
                 }
+                if(skip_headline) {
+                    VLOG_QUERY << "skip is true";
+                    int num_fields_in_csv = range_desc.num_of_columns_from_file;
+                    range_desc.__set_start_offset(num_fields_in_csv);
```
Comment (meegoo) [inline be/src/exec/vectorized/file_scan_node.cpp:301]: why set start offset to num_fields_in_csv?

**Review by meegoo — CHANGES_REQUESTED** [review]

File: be/src/exec/vectorized/file_scan_node.cpp (hunk 2 above)
Comment (zztttt) [reply be/src/exec/vectorized/file_scan_node.cpp:301]: In order to skip the headline if the first line is the attributions name.

**Pull request closed without merging**